app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -492,6 +498,32 @@ private void handleNavIntent() {
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
         setIntent(intent);
+
+        handleIntent(intent);
+    }
+
+    /**
+     *  Handles the incoming intent. If the activity was launched via a deep link,
+     *  passes the uri to handleDeeplink().
+     *
+     *  Added an SDK version check since the minimum SDK supported by AntennaPod is 16
+     *  and Object.equals(Obj1, Obj2) is only supported in API 19+.
+     *
+     * @param intent incoming intent
+     */
+    private void handleIntent(Intent intent) {
```

Comment:
We already have a function named `handleNavIntent`. Could those functions be combined? They seem to do pretty similar things.

Reply:
They could potentially be combined, however, I think keeping them separate is beneficial as they are meant for slightly different purposes.